crates/tedge_command_config/tests/test_set_command.rs

```diff
@@ -0,0 +1,13 @@
+fn set_command_fails_given_device_id() {
+    let command = SetConfigCommand {
+        config_key: ConfigKey::from_str("device.id"),
+        // ...
+    };
+
+    let context = TestContext {}; // We would actually pass in an &dyn ExecutionContext, so to simplify testing.
+    let result = command.execute(&context);
```

Comment:
we can also `assert_eq!(context.stdout_as_string(), "...")`, once we convert every `println!()` to use `context.output(...)`